src/rocks_compaction_scheduler.h

```diff
@@ -63,5 +69,8 @@ namespace mongo {
         // We'll compact the prefix if any operation on the prefix reports more than 50.000
         // deletions it had to skip over (this is about 10ms extra overhead)
         static const int kSkippedDeletionsThreshold = 50000;
+
+        // thread for async execution of range compactions
+        CompactionBackgroundJob* _compactionJob;
```

Comment:
No it does not. It was specifically designed to fulfill this case as well.

`class CompactionBackgroundJob;
std::unique_ptr<CompactionBackgroundJob> _compactionJob;`

But you still need to have the destructor of containing class (RocksCompactionScheduler) defined in the place where CompactionBackgroundJob is fully defined. I'd prefer even explicit reset with the comment:
`RocksCompactionScheduler::~RocksCompactionScheduler() {
  // We need this to avoid incomplete type deletion
  _compactionJob.reset();
}`

Please, let's use `unique_ptr` as it clearly shows the ownership of the object and is a best practice.